packages/framework-provider-aws/src/library/events-adapter.ts

```diff
@@ -88,7 +88,37 @@ export async function storeEvents(
   config: BoosterConfig,
   logger: Logger
 ): Promise<void> {
-  // TODO: Manage query pagination and db.batchWrite limit of 25 operations at a time
+  const batchSize = 25
+  let processedItems = 0
+  let currentBatch = getBatchFromEnvelopes(eventEnvelopes, processedItems, batchSize)
+  while (currentBatch.length != 0) {
+    await persistBatch(logger, currentBatch, config, dynamoDB)
+    processedItems += currentBatch.length
+    currentBatch = getBatchFromEnvelopes(eventEnvelopes, processedItems, batchSize)
+  }
+}
+
+function validBatchSize(eventEnvelopes: EventEnvelope[], processedItems: number, batchSize: number): boolean {
+  return Math.floor((eventEnvelopes.length - processedItems) / batchSize) != 0
+}
+
+function getBatchFromEnvelopes(
+  eventEnvelopes: EventEnvelope[],
+  processedItems: number,
+  batchSize: number
+): EventEnvelope[] {
+  if (!validBatchSize(eventEnvelopes, processedItems, batchSize)) {
+    return eventEnvelopes.slice(processedItems)
+  }
+  return eventEnvelopes.slice(processedItems, processedItems + batchSize)
+}
```

Comment:
Now that I think on it, why not doing it the following way:

1. Separate in chunks
2. Iterate and persist

Something like

```suggestion
  const batches = inChunksOf(25, eventEnvelopes)
  for (const batch of batches) {
    await persistBatch(logger, batch, config, dynamoDB)
  }
}

function inChunksOf<TElement>(n: number, arr: Array<TElement>): Array<TElement> {
  var result = []
  while (arr.length) {
    result.push(arr.splice(0, chunk_size))
  }
  return result
}

function validBatchSize(eventEnvelopes: EventEnvelope[], processedItems: number, batchSize: number): boolean {
  return Math.floor((eventEnvelopes.length - processedItems) / batchSize) != 0
}
```

Reply:
Using @NickSeagull's implementation, with some extra, you could even send all chunks in parallel (not sure if that makes sense TBH):

```typescript
const batchSize = 25
Promise.allSettled(
  inChunksOf(batchSize, eventEnvelopes) // Builds the chunks
    .map((batch) => persistBatch(logger, batch, config, dynamoDB)) // Maps them to promises
)  // Promise.allSettled awaits for all of them

function inChunksOf<TElement>(n: number, arr: Array<TElement>): Array<TElement> {
  let result = []
  while (arr.length) {
    result.push(arr.splice(0, chunk_size))
  }
  return result
}
```

Reply:
@javiertoledo I really like your approach sending all chunks in parallel but I'm not sure about the order because if we are saving all the events in parallel we can't guarantee in which order are they saved in the event store.  What do you think?

Reply:
Good point. Events are always read in order, using the timestamps as a sort key, so if we're generating the timestamps within this loop, then we could easily lose the right order. If we could generate the timestamps before and make the inserts in parallel, I think it won't matter in which order they are inserted. Anyway, it's a micro-optimization and it's hard to predict what will happen as we're also relying on event insertion to trigger event handlers, snapshooting, etc. So 100% agree that it's better to avoid parallelizing this at this time :trollface:

Reply:
Love the Nicks suggestion. Regarding the timestams, they are generated in the `core` so, here, when we already have the event envelopes, it would be save to store in parallel. BUT! 100% agree that we should forget now about that optimization too